framework/validators/IpValidator.php

```diff
@@ -0,0 +1,442 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\validators;
+
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * IpValidator validates that the attribute value is a valid IPv4/IPv6 address or subnet.
+ * May change attribute's value if normalization is enabled.
+ *
+ * @author SilverFire <d.naumenko.a@gmail.com>
+ * @since 2.0.4
+ */
+class IpValidator extends Validator
+{
+    /**
+     * @const integer the length of IPv6 address in bits
+     */
+    const IPV6_ADDRESS_LENGTH = 128;
+
+    /**
+     * @const integer the length of IPv4 address in bits
+     */
+    const IPV4_ADDRESS_LENGTH = 32;
+
+    /**
+     * @const integer is used to change check order by [[isAllowed]]
+     * @see isAllowed
+     */
+    const ORDER_ALLOW_DENY = 0;
+
+    /**
+     * @const integer is used to change check order by [[isAllowed]]
+     * @see isAllowed
+     */
+    const ORDER_DENY_ALLOW = 1;
+
+    /**
+     * @var boolean whether support of IPv6 addresses is enabled
+     */
+    public $ipv6 = true;
+
+    /**
+     * @var boolean whether support of IPv4 addresses is enabled
+     */
+    public $ipv4 = true;
+
+    /**
+     * @var boolean|string whether address can be a CIDR subnet
+     *   boolean - whether an IP address with the CIDR subnet notation will be able to pass the validation
+     *   string  - value 'only' to validate only the addresses with the CIDR
+     */
+    public $subnet = false;
+
+    /**
+     * @var boolean whether to add the prefix with the smallest length (32 for IPv4 and 128 for IPv6)
+     * to an address without it.
+     * Works only when attribute [[subnet]] is not false.
+     * Example: `10.0.1.5` will normalized to `10.0.1.5/32`, `2008:db0::1` will be normalized to `2008:db0::1/128`
+     * @see subnet
+     */
+    public $normalize = false;
+
+    /**
+     * @var boolean|string whether address may have an negation-character at the beginning
+     *   boolean - character "!" will be used
+     *   string - passed character will be used
+     */
+    public $negationChar = false;
+
+    /**
+     * @var boolean whether to expand an IPv6 address to the full notation format
+     */
+    public $expandV6 = false;
+
+    /**
+     * @var int the order of ranges rules. Used by [[isAllowed]]
```

Comment:
Yes, I think it's better to move the concrete description to here because this is the place where users will read the most when they want to figure out how to configure a property.